client/multi-currency-setup/tasks/add-currencies-task/index.js

```diff
@@ -0,0 +1,339 @@
+			[ currencyCode ]: enabled,
+		} ) );
+	};
+
+	// Search component
+	const [ searchText, setSearchText ] = useState( '' );
+	const handleSearchChange = ( event ) => {
+		setSearchText( event.target.value );
+	};
+	const filteredCurrencyCodes = ( ! searchText
+		? availableCurrencyCodes
+		: availableCurrencyCodes.filter( ( code ) => {
+				const { symbol, name } = availableCurrencies[ code ];
+				return (
+					-1 <
+					`${ symbol } ${ code } ${ name }`
+						.toLocaleLowerCase()
+						.indexOf( searchText.toLocaleLowerCase() )
+				);
+		  } )
+	).filter( ( code ) => {
+		// Hide already enabled ones from the search results.
+		return ! hiddenCurrencies.includes( code );
+	} );
+
+	// This state is used for displaying the checked currencies count on the button
+	// and on the description text displayed after adding those checked currencies.
+	useEffect( () => {
+		if ( isActive ) {
+			const activatedCurrenciesCount =
+				Object.values( selectedCurrencies ).filter( Boolean ).length -
+				enabledCurrenciesLength;
+			setActivatedCount(
+				0 < activatedCurrenciesCount ? activatedCurrenciesCount : 0
+			);
+		}
+	}, [ selectedCurrencies, isActive, enabledCurrenciesLength ] );
+
+	const ContinueButton = () => {
+		const checkedCurrencies = useMemo(
+			() =>
+				Object.entries( selectedCurrencies )
+					.map( ( [ currency, enabled ] ) => enabled && currency )
+					.filter( Boolean ),
+			[]
+		);
+
+		const handleContinueClick = useCallback( () => {
+			setStatus( 'pending' );
+			checkedCurrencies.sort();
+			submitEnabledCurrenciesUpdate( checkedCurrencies );
+			setStatus( 'resolved' );
+			setCompleted(
+				{
+					initialCurrencies: enabledCurrencies,
+				},
+				'multi-currency-settings'
+			);
+		}, [ checkedCurrencies ] );
+
+		return (
+			<Button
+				isBusy={ 'pending' === status }
+				disabled={ 'pending' === status || 1 > activatedCount }
+				onClick={ handleContinueClick }
+				isPrimary
+			>
+				{ 0 === activatedCount
+					? __( 'Add Currencies', 'woocommerce-payments' )
+					: sprintf(
+							_n(
+								'Add %s currency',
+								'Add %s currencies',
+								activatedCount,
+								'woocommerce-payments'
+							),
+							activatedCount
+					  ) }
+			</Button>
+		);
+	};
+
+	const displayCurrencyCheckbox = ( code ) =>
+		hiddenCurrencies.includes( code ) ? null : (
+			<EnabledCurrenciesModalCheckbox
+				key={ 'currency-checkbox-' + availableCurrencies[ code ].id }
+				checked={ selectedCurrencies[ code ] }
+				onChange={ handleChange }
+				currency={ availableCurrencies[ code ] }
+			/>
+		);
+
+	return (
+		<WizardTaskItem
+			className="add-currencies-task"
+			title={ __( 'Add currencies', 'woocommerce-payments' ) }
+			visibleDescription={ sprintf(
+				_n(
+					'%s currency added',
+					'%s currencies added',
+					activatedCount,
+					'woocommerce-payments'
+				),
+				10 > activatedCount
+					? _.capitalize( numberWords[ activatedCount ] )
+					: activatedCount
```

Comment:
🤔 If I'm not wrong this will not be internationalized. Cause the constant `numberWords` will always show `one` for `1` in any language. Can you confirm it?

Reply:
True, nice catch! I'll wrap it inside a translation method! 👍🏼

Reply:
Addressed in a592b44. Thanks for the catch!